src/GenericRestClient.ts

```diff
@@ -51,12 +51,7 @@ export class GenericRestClient {
             options.sendData = objToPost;
         }
 
-        const promise = this._blockRequestUntil(options);
-        if (!promise) {
-            return this._performApiCallInternal(apiPath, action, options);
-        }
-
-        return promise.then(() => this._performApiCallInternal(apiPath, action, options));
+        return this._performApiCallInternal(apiPath, action, options);
```

Comment:
You don't need a separate `_performApiCallInternal` any more. Merge with `_performApiCall`.